tests/comdb2_files.test/runit

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+bash -n "$0" | exit 1
+
+dbnm=$1
+dir=$DBDIR
+
+function runtest {
+	echo "restricted blahblah" > $dir/foo
+	chmod 000 $dir/foo
```

Comment:
From the man page:

access() checks whether the calling process can access the file pathname.  If pathname is a symbolic link, it is dereferenced.

So I think you may be able to work this around by creating a symlink to a file that does not exist.

Reply:
Sounds like a plan!

Reply:
As a follow up: What do you think is the right thing to do (1) when a file doesn't exist and (2) when a file is restricted?

Right now a non-existent file causes a select from `comdb2_files` to fail. This could cause an import (or some other important process that uses `comdb2_files`) to fail because of a broken symlink. We could skip these files with a log message, or we could fail and expect the user to address any bad files before attempting the import. 

I think that restricted files should be skipped since restricted files in the database directory are expected.

Reply:
I'd skip a file with a log message, in both cases.

Reply:
I missed that [stat results are used to check if the entry is a directory](https://github.com/bloomberg/comdb2/pull/4307/files#diff-e5e5f17584917389f36bac6456db718edf2c435685b06bf7b08f05e2c506efc5R358). I opted to use both `access()` and `stat()` instead of just using `stat()` because I wanted to avoid parsing the permissions in the `stat()` result and checking if the running process is op or not. 

I decided to skip files that are read-restricted and to fail on all other `stat()` and `access()` errors. I did this because if we're running recovery with the result-set of `comdb2_files()`, then skipping a file that is required for recovery could result in a corrupt database. I'm assuming that files that are read-restricted to the database process will never be required in recovery, so it's okay to skip those.

Let me know what you think!

Reply:
The test has 3 cases: 
1) check that a select from `comdb2_files` succeeds when there's a read-restricted file
2) check that a select from `comdb2_files` succeeds when there's a read-restricted directory
3) check that a select from `comdb2_files` fails when there's a broken symlink.

The first 2 cases aren't very meaningful when comdb2 is run as root. I'm not sure how to work around this though.